sunbeam-python/sunbeam/commands/microceph.py

```diff
@@ -191,29 +190,34 @@ def get_unpartitioned_disks(self) -> list:
                     APPLICATION, self.machine_id, self.model
                 )
             )
-            _, unpartitioned_disks = run_sync(
+            self.osd_disks, self.unpartitioned_disks = run_sync(
                 list_disks(self.jhelper, self.model, unit.entity_id)
             )
-            unpartitioned_disks = [disk.get("path") for disk in unpartitioned_disks]
+            self.unpartitioned_disks = [
+                disk.get("path") for disk in self.unpartitioned_disks
+            ]
             # Remove duplicates if any
-            unpartitioned_disks = list(set(unpartitioned_disks))
-            if OSD_PATH_PREFIX in unpartitioned_disks:
-                unpartitioned_disks.remove(OSD_PATH_PREFIX)
+            self.unpartitioned_disks = list(set(self.unpartitioned_disks))
+            if OSD_PATH_PREFIX in self.unpartitioned_disks:
```

Comment:
I don't really get the need for that, were there instances where the unpartitioned_disks list got an element with only OSD_PATH_PREFIX ?

Reply:
Agree, not sure why i have added that logic too a year ago. Also removed the line which filters the duplicates, having duplicates should throw an error so that user can take action.